src/Facade/Services/ShortagesReportFacadeService.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace Linn.Purchasing.Facade.Services
+{
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Reporting.Models;
+    using Linn.Common.Reporting.Resources.ReportResultResources;
+    using Linn.Purchasing.Domain.LinnApps.Reports;
+    using Linn.Purchasing.Resources.RequestResources;
+
+    public class ShortagesReportFacadeService : IShortagesReportFacadeService
+    {
+        private readonly IShortagesReportService domainService;
+
+        private readonly IBuilder<ResultsModel> resultsModelResourceBuilder;
+
+        public ShortagesReportFacadeService(
+            IShortagesReportService domainService,
+            IBuilder<ResultsModel> resultsModelResourceBuilder)
+        {
+            this.domainService = domainService;
+            this.resultsModelResourceBuilder = resultsModelResourceBuilder;
+        }
+
+        public IResult<IEnumerable<ReportReturnResource>> GetReport(ShortagesReportRequestResource options)
+        {
+            var purchaseLevel = int.Parse(options.PurchaseLevel);
```

Comment:
I would make options.PurchaseLevel an int in the resource so you don't have to parse it.

Reply:
This was a thing that was tripping me a little as the table has it marked as a string. As far as I can tell it will always be an integer. Would cause the query to blow up as it would say that it could not convert from a string to an int. Happy to change it but how could I get around this at the query level?